packages/wallets/fractal/src/adapter.ts

```diff
@@ -0,0 +1,209 @@
+import base58 from 'bs58';
+import {
+    SendTransactionOptions,
+    TransactionOrVersionedTransaction,
+    WalletError,
+    WalletNotConnectedError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import type { WalletName } from '@solana/wallet-adapter-base';
+import {
+    BaseSignerWalletAdapter,
+    WalletReadyState,
+    WalletConnectionError,
+    WalletPublicKeyError,
+    WalletNotReadyError,
+} from '@solana/wallet-adapter-base';
+import { Message, Transaction } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import {
+    ConnectionManager,
+    Platform,
+    PopupEvent,
+    assertPayloadIsSolanaWalletAdapterApproved,
+    DEFAULT_POPUP_HEIGHT_PX,
+    assertPayloadIsTransactionSignatureNeededResponsePayload,
+} from '@fractalwagmi/popup-connection';
+import type { TransactionSignatureNeededPayload } from '@fractalwagmi/popup-connection';
+import uuid from 'uuid';
+
+const FRACTAL_DOMAIN_HTTPS = 'https://fractal.is';
+const APPROVE_PAGE_URL = `${FRACTAL_DOMAIN_HTTPS}/wallet-adapter/approve`;
+const SIGN_PAGE_URL = `${FRACTAL_DOMAIN_HTTPS}/wallet-adapter/sign`;
+const MIN_POPUP_HEIGHT_PX = DEFAULT_POPUP_HEIGHT_PX;
+const MAX_POPUP_WIDTH_PX = 850;
```

Comment:
The `@solana/wallet-adapter-fractal` package should be as dumb and thin as possible, and have no external deps. `@fractalwagmi/popup-connection` and `uuid` must be imported as types only here (`uuid` may not be needed at all). All loadable adapters must load any external SDK asynchronously in the `connect` function. Here's how the adapters for [Torus](https://github.com/solana-labs/wallet-adapter/blob/master/packages/wallets/torus/src/adapter.ts#L76-L130) and [WalletConnect](https://github.com/solana-labs/wallet-adapter/blob/e0f88df48c26bfd69b9ae6f34b5c1439f926bdfd/packages/wallets/walletconnect/src/adapter.ts#L64-L108) do it. The `uuid` dependency and the constants initialized below should all be moved to a package you maintain, e.g. `@fractalwagmi/solana-wallet-adapter` and imported async as a single dependency in this package with minimal setup in the `connect` function.

Reply:
Got it, thanks! I'll make those changes and follow up.

I noticed you didn't mention the `bs58` module but I am loading it as an external dep. Should I also load that one asynchronously?

Reply:
Nevermind on my last comment, when I move everything into the async-loaded lib, bs58 will go along with it so this package should no longer depend on it.

Reply:
All good! Yeah, good intuition. Also, bs58 is a special case. It gets a free pass because it's a hard dependency of web3.js Pubkey (and other stuff).

Reply:
Gotcha. ok, I've moved over basically all of the implementation into `@fractalwagmi/solana-wallet-adapter` and cleaned up this `adapter.ts` accordingly by forking over the Torus wallet. PTAL. Thanks!